trade-finance-manager-ui/templates/utilisation-reports/_macros/premium-payments-table.njk

```diff
@@ -3,8 +3,109 @@
+  {% set paymentsReceived = feeRecordPaymentGroup.paymentsReceived %}
+  {% set totalPaymentsReceived = feeRecordPaymentGroup.totalPaymentsReceived %}
+  {% set status = feeRecordPaymentGroup.status %}
+  {% set displayStatus = feeRecordPaymentGroup.displayStatus %}
+  {% set checkboxId = feeRecordPaymentGroup.checkboxId %}
+  {% set isChecked = feeRecordPaymentGroup.isChecked %}
+
+  {% for feeRecord in feeRecords %}
+    {% set isFirstRow = loop.index === 1 %}
+    {% set isLastRow = loop.index === feeRecords.length %}
+
+    <tr
+      class="govuk-table__row"
+      data-cy="premium-payments-table-row--feeRecordId-{{ feeRecord.id }}"
+    >
+      <th
+        scope="row"
+        class="govuk-table__header govuk-!-font-weight-regular{{ " no-border" if not isLastRow }}"
+      >
+        {{ feeRecord.facilityId }}
+      </th>
+      <td class="govuk-table__cell{{ " no-border" if not isLastRow }}">
+        {{ feeRecord.exporter }}
+      </td>
+      <td class="govuk-table__cell govuk-table__cell--numeric{{ " no-border" if not isLastRow }}">
+        {{ feeRecord.reportedFees }}
+      </td>
+      <td class="govuk-table__cell govuk-table__cell--numeric{{ " no-border" if not isLastRow }}">
+        {{ feeRecord.reportedPayments }}
+      </td>
+
+      {% if isFirstRow %}
+        <td
+          class="govuk-table__cell govuk-table__cell--numeric{{ " no-border" if not isLastRow }}"
+          data-sort-value="{{ totalReportedPayments.dataSortValue }}"
+        >
+          {{ totalReportedPayments.formattedCurrencyAndAmount }}
+        </td>
+      {% elif isLastRow %}
+        <td class="govuk-table__cell" data-sort-value="{{ totalReportedPayments.dataSortValue }}"></td>
+      {% else %}
+        <td class="govuk-table__cell no-border" data-sort-value="{{ totalReportedPayments.dataSortValue }}"></td>
+      {% endif %}
+      
+      <td class="govuk-table__cell govuk-table__cell--numeric{{ " no-border" if not isLastRow }}">
+        {% if isFirstRow and paymentsReceived %}
+          <ul class="payments-list">
+            {% for paymentReceived in paymentsReceived %}
+              <li class="payments-list-item">
+                <a href="#">{{ paymentReceived }}</a>
+              </li>
+            {% endfor %}
+          </ul>
+        {% endif %}
+      </td>
+
+      {% if isFirstRow and totalPaymentsReceived %}
+        <td
+          class="govuk-table__cell govuk-table__cell--numeric{{ " no-border" if not isLastRow }}"
+          data-sort-value="{{ totalPaymentsReceived.dataSortValue }}"
+        >
+          {{ totalPaymentsReceived.formattedCurrencyAndAmount }}
+        </td>
+      {% elif isLastRow %}
+        <td class="govuk-table__cell" data-sort-value="{{ totalPaymentsReceived.dataSortValue }}"></td>
+      {% else %}
+        <td class="govuk-table__cell no-border" data-sort-value="{{ totalPaymentsReceived.dataSortValue }}"></td>
+      {% endif %}
```

Comment:
Similarly can this be flattened so the if else is just on the inner content of the cell?